spec/version.dd

```diff
@@ -537,6 +538,127 @@ INT!(17) c;  // error, static assert trips
 	)
 	)
 
+$(H2 $(LNAME2 staticforeach, Static Foreach))
+
+$(GRAMMAR
+$(GNAME StaticForeach):
+    $(D static) $(GLINK2 statement, AggregateForeach)
+    $(D static) $(GLINK2 statement, RangeForeach)
+
+$(GNAME StaticForeachDeclaration):
+    $(GLINK StaticForeach) $(GLINK2 attribute, DeclarationBlock)
+    $(GLINK StaticForeach) $(D :) $(GLINK2 module, DeclDefs)$(OPT)
+
+$(GNAME StaticForeachStatement):
+    $(GLINK StaticForeach) $(GLINK2 statement, NoScopeNonEmptyStatement)
+)
+
```

Comment:
Perhaps you should specify what happens with Ddoc when `StaticForeachDeclaration` declares functions preceded by a ddoc comment. Is documentation generated for each function, or is there only once instance. With `static if` I think there's a trick that goes along the lines of `static if (condition || generatingDoc) /** This function is available iff ... */ void foo();`. Another question is whether you can generate documentation and mix it in in for each declaration (e.g. declare an overload set where each function has the same documentation, save for the parameters section).